src/extensions/png/mod-lodepng.c

```diff
@@ -44,35 +53,279 @@
 
 #include "tmp-mod-lodepng-first.h"
 
-// !!! This didn't really have anywhere to go.  It came from %host-core.c,
-// and it's not part of the historical PNG code, but apparently Saphirion
-// found a problem with that in terms of saving (saving only?) which they
-// added in lodepng for.  This is unfortunate as lodepng repeats deflate
-// code already available in Zlib.
+
+//=//// CUSTOM SERIES-BACKED MEMORY ALLOCATOR /////////////////////////////=//
+//
+// LodePNG allows for a custom allocator, but it assumes the pointer it is
+// given is where it will write data...so it can't be given something like
+// a REBSER* which needs BIN_HEAD() or BIN_AT() to dereference it.  But we
+// would like to avoid the busywork of copying data between malloc()'d buffers
+// and REBSER data which can be given into userspace.
+//
+// A trick is used here where a series is allocated that is slightly larger
+// than the requested data size...just large enough to put a pointer to the
+// series itself at the head of the memory.  Then, the memory right after
+// that pointer is given back to LodePNG.  The series pointer can then be
+// found again by subtracting sizeof(REBSER*) from the client pointer.
+//
+// lodepng contains prototypes for these functions, and expects them to
+// be defined somewhere if you #define LODEPNG_NO_COMPILE_ALLOCATORS
+// (this is specified in the extension compiler flag settings)
 //
-// It is used as an override for the encoder from R3-Alpha, which is found
-// in %u-png.c as ENCODE-PNG.
+//=////////////////////////////////////////////////////////////////////////=//
 
+void* lodepng_malloc(size_t size)
+{
+    return Rebol_Malloc(size);
+}
+
+void* lodepng_realloc(void* ptr, size_t new_size)
+{
+    return Rebol_Realloc(ptr, new_size);
+}
+
+void lodepng_free(void* ptr)
+{
+    return Rebol_Free(ptr);
```

Comment:
Yup.  So do you have any idea what's going on with this build?  It stopped working a bit ago, I thought it might be temporary, but doesn't seem to be...

https://travis-ci.org/metaeducation/ren-c/jobs/260324499

Incidentally, I think it would be nice if we kept a mixture of Linux on there, some new and some old...so deliberately set a few of them to be on the old platform.